### PR TITLE
Creation of receiveHeaders function

### PR DESCRIPTION
**Pull request by pamgonzalez**
It includes a new function in bridgeMethods called receiveHeader, which will be public and receives a single header as an input parameter, being able to return errors.
This function can be used from RSKIP200

In turn, receiveHeaders is no longer public with this same RSKIP.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeMethods.java [hunk 1]
```diff
@@ -485,6 +485,17 @@
...
             false
     ),
+    RECEIVE_HEADER(
+            CallTransaction.Function.fromSignature(
+                    "receiveHeader",
+                    new String[]{"bytes"},
+                    new String[]{"int256"}
+            ),
+            fixedCost(22_000L),  // TODO: calculate gas cost.
+            (BridgeMethodExecutorTyped) Bridge::receiveHeader,
```
Comment (guidohernan93) [inline rskj-core/src/main/java/co/rsk/peg/BridgeMethods.java:495]: Is that cast needed?

Comment (pamgonzalez) [reply rskj-core/src/main/java/co/rsk/peg/BridgeMethods.java:495]: Probably not. It has all the functions that are not void and it is a good way in view of being able to differentiate it. Anyway I can review it.

**Comment by pamgonzalez** [comment]
pipeline:run

File: rskj-core/src/test/java/co/rsk/peg/BridgeTest.java [hunk 2]
```diff
@@ -193,13 +197,125 @@ public void registerBtcCoinbaseTransaction_after_RSKIP143_activation_null_data()
...
+        ActivationConfig activations = spy(ActivationConfigsForTest.genesis());
+        doReturn(true).when(activations).isActive(eq(RSKIP200), anyLong());
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        Bridge bridge = spy(getBridgeInstance(bridgeSupportMock, activations));
+
+        byte[] data = Bridge.RECEIVE_HEADER.encode(new Object[]{});
+
+        Assert.assertNull(bridge.execute(data));
+        verify(bridge, never()).receiveHeaders(any(Object[].class));
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeTest.java:237]:
```suggestion
        verify(bridge, never()).receiveHeader(any(Object[].class));
```

File: rskj-core/src/test/java/co/rsk/peg/BridgeTest.java [hunk 3]
```diff
@@ -193,13 +197,125 @@ public void registerBtcCoinbaseTransaction_after_RSKIP143_activation_null_data()
...
+        verifyZeroInteractions(bridgeSupportMock);
+    }
+
+    @Test
+    public void receiveHeader_Ok() throws VMException {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeTest.java:242]:
```suggestion
    public void receiveHeader_after_RSKIP200_Ok() throws VMException {
```

File: rskj-core/src/main/java/co/rsk/peg/Bridge.java [hunk 4]
```diff
@@ -436,6 +437,20 @@ public void receiveHeaders(Object[] args) throws VMException {
         }
     }
 
+    public int receiveHeader(Object[] args) throws VMException {
+        logger.trace("receiveHeader");
+
+        BtcBlock header = bridgeConstants.getBtcParams().getDefaultSerializer().makeBlock((byte[]) args[0]);
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/Bridge.java:443]: you are missing the length validation (see receiveHeaders)

**Review by josedahlquist — CHANGES_REQUESTED** [review]
The changes look good but we need to add the length validation

**Review by marcos-iov — APPROVED** [review]

**Review by josedahlquist — APPROVED** [review]

**Pull request merged**